File: Interviews/LeetCode/Daily-Problems/2022/October/Day-29/2136.cpp
class Solution {
public:
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
        int n = plantTime.size();
        vector<int> indices(n);
        iota (indices.begin(), indices.end(), 0);

        sort (indices.begin(), indices.end(), [&](int a, int b) {
            return growTime[a] > growTime[b];
        });

        int ans = 0, sum = 0;
        for (int i = 0; i < n; i++) {
            sum += plantTime[indices[i]];
            ans = max(ans, sum + growTime[indices[i]]);
        }

        return ans;
    }
};

File: Interviews/LeetCode/Topic-Wise/Greedy/Hard/Earliest-Possible-Day-of-Full-Bloom/2136.py
class Solution:
    def earliestFullBloom(self, plantTime: List[int], growTime: List[int]) -> int:
        indices = list (range (0, len(plantTime)))
        indices.sort (key = lambda x: -growTime[x])
        ans, cursum = 0, 0
        for i in indices:
            cursum += plantTime[i]
            ans = max(ans, cursum + growTime[i])
        
        return ans